ephios/core/views/workinghour.py

```diff
@@ -23,15 +24,15 @@
 
 
 class WorkingHourPermissionMixin:
-    def setup(self, request, *args, **kwargs):
-        result = super().setup(request, *args, **kwargs)
-        can_grant_for_group_ids = get_objects_for_user(
+    @cached_property
+    def can_grant(self):
+        """Return whether the current request user can grant working hours to the target user (which might be themselves)"""
+        if self.request.user.is_anonymous:
+            return False
+        can_grant_for_groups = get_objects_for_user(
             self.request.user, "decide_workinghours_for_group", klass=Group
-        ).values_list("id", flat=True)
-        self.can_grant = (
-            self._get_target_user().groups.filter(id__in=can_grant_for_group_ids).exists()
         )
-        return result
+        return self._get_target_user().groups.filter(id__in=can_grant_for_groups).exists()
```

Comment:
Problem was that when `request.user` is anonoumous, the `get_objects_for_user` was failing because we don't use guardians anonymous user feature. But at the view's setup stage, the redirect to login did not occur yet. I fixed it first for the detail view by making it a property only computed after the view's disposition stage in which an anonymous user might be redirected to login. But that didn't help with the views that require users to `can_grant` to have permission in the first place. That's when I added the 

```
if self.request.user.is_anonymous:
    return False
```

which I think would be enough in the first place, though I do like it more this way either way.